testsuite/tests/docstrings/docstrings.ml

```diff
@@ -0,0 +1,575 @@
+        there is a blank line before titi, but is kept
+        as a comment in the class. *)
+
+    val titi : string
+
+    (** Comment for method toto *)
+    method toto : string
+
+    (** Comment for method m *)
+    method m : float -> int
+  end
+
+  (** The comment for the class type my_class_type *)
+  class type my_class_type = object
+    (** The comment for variable x. *)
+    val mutable x : int
+
+    (** The commend for method m. *)
+    method m : int -> int
+  end
+
+  (** The comment for module Foo *)
+  module Foo : sig
+    (** The comment for x *)
+    val x : int
+
+    (** A special comment that is kept but not associated to any element *)
+  end
+
+  (** The comment for module type my_module_type. *)
+  module type my_module_type = sig
+    (** The comment for value x. *)
+    val x : int
+
+    (** The comment for module M. *)
+    module M : sig
+      (** The comment for value y. *)
+      val y : int
+
+      (* ... *)
+    end
+
+  end
+
+end = struct
+
+  (** The comment for function f *)
+  let f x y = x + y
+
+  (** This comment is not attached to any element since there is another
+      special comment just before the next element. *)
+
+  (** Comment for exception My_exception, even with a simple comment
+      between the special comment and the exception.*)
+  (* A simple comment. *)
+  exception My_exception of (int -> int) * int
+
+  (** Comment for type weather  *)
+  type weather =
+    | Rain of int (** The comment for constructor Rain *)
+    | Sun (** The comment for constructor Sun *)
+
+  (** The comment for type my_record *)
+  type my_record = {
+    foo : int ;    (** Comment for field foo *)
+    bar : string ; (** Comment for field bar *)
+  }
+
+  (** The comment for class my_class *)
+  class my_class = object
+    (** A comment to describe inheritance from cl *)
+    inherit cl
+
+    (** The comment for the instance variable tutu *)
+    val mutable tutu = "tutu"
+
+    (** The comment for toto *)
+    val toto = 1
+    val titi = "titi"
+    (** Ambiguous comment on both titi and toto *)
+    method toto = tutu ^ "!"
+
+    (** The comment for method m *)
+    method m (f : float) = 1
+  end
+
+  (** The comment for class type my_class_type *)
+  class type my_class_type = object
+    (** The comment for the instance variable x. *)
+    val mutable x : int
+
+    (** The comment for method m. *)
+    method m : int -> int
+  end
+
+  (** The comment for module Foo *)
+  module Foo = struct
+    (** The comment for x *)
+    val x : int
+    (** Another comment for x *)
+  end
+
+  (** The comment for module type my_module_type. *)
+  module type my_module_type = sig
+    (* Comment for value x. *)
+    val x : int
+    (* ... *)
+  end
+
+end;;
+[%%expect {|
+
+module Manual :
+  sig
+    [@@@ocaml.text
+      " Special comments can be placed between elements and are kept\n      by the OCamldoc tool, but are not associated to any element.\n      @-tags in these comments are ignored."]
```

Comment:
is there a way to use line breaks in the `expect` result here that is portable and correct, and increases readability of the test file?

Reply:
Without changing how `-dsource` prints things, any such change would be undone when promoting the result of the test.

Reply:
So, alternative question: is there a way for `-dsource` to print this in a nicer way? (This would be a separate PR.)

Reply:
I think that 

```
"  foo
  bar"
```
is not portable due to \n vs. \r\n concerns,

```
"  foo\n\
  bar"
```

does not work as spaces at the beginning of the line are ignored, but that one could use


```
"  foo\n  \
   bar"
```

pushing indentation of a line on the previous line.